core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java

```diff
@@ -101,7 +101,13 @@
     public JnlpSlaveAgentProtocol4() throws KeyStoreException, KeyManagementException, IOException {
         // prepare our local identity and certificate
         X509Certificate identityCertificate = InstanceIdentityProvider.RSA.getCertificate();
+        if (identityCertificate == null) {
+            throw new KeyStoreException("no X509Certificate found; perhaps instance-identity is missing or too old");
```

Comment:
Maybe it should explicitly mention the protocol. Just to make the error more diagnosable